### ISSUE TITLE
player_stop is buggy

### ISSUE THREAD

**Issue by kerizane** [Tuxemon/Tuxemon#583]
If you go to route2 and talk to Misa you will be able to walk around after triggering her last piece of dialog, this is despite us calling "player_stop".

```
  <object id="134" name="spilledthebeans" type="event" x="48" y="48" width="16" height="16">
   <properties>
    <property name="act1" value="player_stop"/>
    <property name="act2" value="wait 1"/>
    <property name="act3" value="translated_dialog leavemealone"/>
    <property name="act4" value="player_resume"/>
    <property name="cond1" value="is player_at"/>
    <property name="cond2" value="is player_facing left"/>
    <property name="cond3" value="is button_pressed K_RETURN"/>
    <property name="cond4" value="is variable_set leavemealone:yes"/>
   </properties>
  </object>
```

**Comment by kerizane** [comment]
Doesn't work:
`  <object id="134" name="spilledthebeans" type="event" x="48" y="48" width="16" height="16">
   <properties>
    <property name="act1" value="player_stop"/>
    <property name="act2" value="wait 1"/>
   </properties>
  </object>
`
Works:
`  <object id="134" name="spilledthebeans" type="event" x="48" y="48" width="16" height="16">
   <properties>
    <property name="act1" value="player_stop"/>
   </properties>
  </object>
`
Works
`
  <object id="134" name="spilledthebeans" type="event" x="48" y="48" width="16" height="16">
   <properties>
    <property name="act1" value="player_stop"/>
    <property name="act2" value="create_npc misa,2,3,misa,stand"/>
   </properties>
  </object>
`

**Comment by kerizane** [comment]
Looks like calling  "wait" somehow stops it from working.

**Comment by Qiangong2** [comment]
If I remember correctly, player_stop just makes the player stop moving for a second. That's why wait is there. To keep you in your place

**Comment by kerizane** [comment]
You're partly right:

    def stop_player(self):
        """ Reset controls and stop player movement at once.  Do not lock controls

        :return:
        """
        self.wants_to_move_player = None
        self.game.release_controls()
        self.player1.cancel_movement()

**Comment by kerizane** [comment]
What is meant to happen in that last bit?

- Player talks to her
- Player can't move
- Misa says "We used to play here all the time"
- Player can move again.

?

**Comment by Qiangong2** [comment]
Yes. That's supposed to happen. Also, you shouldn't be able to spam click and open the dialog boxes multiple times

**Comment by Qiangong2** [comment]
I'd be fine implementing player_lock. It would simplify events quite a bit

**Comment by bitcraft** [comment]
Events from the dialog might be passed down to the world state.  If you imagine the game as a stack of state, each on top of the other, the top state gets inputs first, then chooses to send the down if it is not interested.  During my last big patch I cleaned up some hacks that cleared inputs and introduced this bug.  Thanks for reporting it, Ill take a look.

There is already a lock method for the game state.  We should have an EventAction as well.  Its a common function to use in pokemon scripting.  We/I should make sure inputs from states are being processed correctly. 
```python
    ####################################################
    #                Player Movement                   #
    ####################################################
    def lock_controls(self):
        """ Prevent input from moving the player
        :return:
        """
        self.allow_player_movement = False
```

**Comment by bitcraft** [comment]
actually, on [re]reviewing this thread, the player stop/resume actions are basically lock/unlock, but by another name.  IMO stop should just be stopping movement, lock/unlock should prevent movement.  resume should only apply if the NPC/player has a path applied to them, and they are currently folling it, or pathfinding to a spot.

**Comment by bitcraft** [comment]
fixed here 3acf21216fbcaa7a2023a69fc7ec50a724d93d07
there is already lock/unlock functions.  fixed by locking during the combat event, then unlocking when the world state resumes.  @kerizane please test when you get a moment
